### PR TITLE
Update ConnectWalletViewController

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #2048  #2049

### CODE REVIEW

File: Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift [hunk 1]
```diff
@@ -30,147 +38,159 @@ class ConnectWalletViewController: UITableViewController {
...
-    private func enterName(for session: Session) {
-        // get the address of the connected wallet
-        guard let walletInfo = session.walletInfo,
-              let address = walletInfo.accounts.first.flatMap(Address.init) else {
-                  App.shared.snackbar.show(error: GSError.WCConnectedKeyMissingAddress())
-            return
-        }
+        searchController.searchResultsUpdater = self
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.searchBar.placeholder = "Search"
```
Comment (elgatovital) [inline Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift:56]: Search -> Search app?

File: Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift [hunk 2]
```diff
@@ -30,147 +38,159 @@ class ConnectWalletViewController: UITableViewController {
...
-                waitingForSession = true
                 show(WalletConnectQRCodeViewController.create(code: connectionURI), sender: nil)
+            default:
+                let wallet = sections[indexPath.section].rows[indexPath.row]
+                if wallet.installed {
+                    // TODO: Create connection request
+                } else if let storeURL = wallet.appStoreLink {
+                    UIApplication.shared.open(storeURL, options: [:], completionHandler: nil)
+                } else {
+                    App.shared.snackbar.show(message: "Wallet not installed and store link is missing")
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift:121]: in this case we can try opening `homepage`

File: Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift [hunk 3]
```diff
@@ -30,147 +38,159 @@ class ConnectWalletViewController: UITableViewController {
...
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        if sections[section].title.isEmpty { return 0 }
         return BasicHeaderView.headerHeight
     }
+
+    func bindData() {
+        wallets = walletsSource.wallets(searchTerm)
+        if searchTerm == nil {
+            sections = [.init(type: .qrCode, title: "", rows: []),
+                        .init(type: .installedWallets, title: "ON THIS DEVICE", rows: wallets.filter { $0.installed }),
```
Comment (DmitryBespalov) [inline Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift:147]: what if this section has 0 rows?

**Review by DmitryBespalov — APPROVED** [review]
Looks good! 🚀 I'm excited!

File: Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift (hunk 3 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/ConnectWalletViewController.swift:147]: Actually You are right.

**Pull request merged**